### PR TITLE
Add io.github.nokse22.Exhibit

### PR DESCRIPTION
**Pull request by Nokse22**
**Exhibit** is a 3d model viewer for the GNOME desktop.

### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by Nokse22** [comment]
bot, build io.github.nokse22.Exhibit

File: io.github.nokse22.Exhibit.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            {
+	    "name": "python3-f3d",
+	    "buildsystem": "simple",
+	    "build-commands": [
+		"pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"f3d\" --no-build-isolation"
+	    ],
+	    "sources": [
+		{
+		    "type": "file",
+		    "url": "https://files.pythonhosted.org/packages/c0/84/56b87aa035fce479e9ec4ac1af1210a273ebede9c2df1b1d06e33f7cc63d/f3d-2.4.0-cp310-cp310-manylinux_2_28_x86_64.whl",
```
Comment (hfiguiere) [inline io.github.nokse22.Exhibit.json:33]: so who will this work on aarch64?

Also why do you need 4 version of the same arch? Python 3.11 is what's in the runtime.

File: io.github.nokse22.Exhibit.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
+    "id" : "io.github.nokse22.Exhibit",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "46",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "exhibit",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=x11",
```
Comment (hfiguiere) [inline io.github.nokse22.Exhibit.json:9]: this uses Gtk4. Why X11 only?

Comment (Nokse22) [reply io.github.nokse22.Exhibit.json:9]: Unfortunately there is an issue between the library that I use to render the 3d models and Gtk, so it only works in X11 for some reason, still don't know how to fix it.

File: io.github.nokse22.Exhibit.json (hunk 1 above)
Comment (Nokse22) [reply io.github.nokse22.Exhibit.json:33]: I haven't tested it on aarch64, but the library should work on Arm.

Will fix the packages.

**Comment by Nokse22** [comment]
How do I disable aarch? Thank you

File: io.github.nokse22.Exhibit.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            {
+	    "name": "python3-f3d",
+	    "buildsystem": "simple",
+	    "build-commands": [
+		"pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"f3d\" --no-build-isolation"
+	    ],
+	    "sources": [
+		{
+		    "type": "file",
+		    "url": "https://files.pythonhosted.org/packages/2c/fe/2e5e38b743f594284932fb2f080e81668b9544dc9fd4d909a5f13ed424aa/f3d-2.4.0-cp39-cp39-manylinux_2_28_x86_64.whl",
```
Comment (hfiguiere) [inline io.github.nokse22.Exhibit.json:33]: Python in the runtime is 3.11

Comment (hfiguiere) [reply io.github.nokse22.Exhibit.json]: but why not building the package from source.

File: io.github.nokse22.Exhibit.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	    ],
+	    "sources": [
+		{
+		    "type": "file",
+		    "url": "https://files.pythonhosted.org/packages/1f/65/60eaae1f9c648ebc68797a73f8417a722ef3c3571ac9d44b44459143ef48/f3d-2.4.0-cp311-cp311-manylinux_2_28_x86_64.whl",
+		    "sha256": "2e14129e7d9432a5d7ebda132077b761c998b80c5605ed4fa351d2d802c0402e"
+		}
+	    ]
+	},
+        {
```
Comment (bbhtt) [inline io.github.nokse22.Exhibit.json:38]: Indentation is broken here.

**Comment by Nokse22** [comment]
I now build the package from source.

I also fixed the wayland issue, but i had to add `"--env=GDK_DEBUG=gl-egl"` to have EGL on X11, this way it works both in wayland and x11.

**Comment by Nokse22** [comment]
bot, build io.github.nokse22.Exhibit

**Comment by Nokse22** [comment]
bot, build io.github.nokse22.Exhibit

**Comment by bbhtt** [comment]
bot, build io.github.nokse22.Exhibit

**Pull request closed without merging**